refactor(env-vars): auth environment validation

naftis opened the pull request:
#5932 

Refactors auth to use envalid and enforce setting environment variables in production.

naftis committed:
refactor: use envalid in auth

naftis commented on packages/auth/src/constants.ts:
```diff
@@ -8,46 +8,6 @@
-  ? parseInt(process.env.CONFIG_SYSTEM_TOKEN_EXPIRY_SECONDS, 10)
-  : 600
 
 export const WEB_USER_JWT_AUDIENCES = [
```
Should we aim to move these leftover constants closer to where they are needed? These are related to creating the JWT so they should be defined [here](https://github.com/opencrvs/opencrvs-core/blob/62df49f96575d861bb199e48334282a65a447e87/packages/auth/src/features/authenticate/service.ts)

rikukissa replied:
@naftis yes I think so, if the are not configurable

naftis replied:
I'll leave them here for now as the `authenticate/service.ts` also isn't the best place, as many of the other handlers/services also refer to `authenticate/service.ts`, we should rather have an `authenticate.ts` which is referred by the services/handlers. I remember refactoring this in the MOSIP pre-work so lets touch it when I get back to that work.

rikukissa commented on packages/auth/src/environment.ts:
```diff
@@ -0,0 +1,33 @@
+  CLIENT_APP_URL: url({ devDefault: 'http://localhost:3000/' }),
+  CERT_PRIVATE_KEY_PATH: str({ devDefault: '../../.secrets/private-key.pem' }),
+  CERT_PUBLIC_KEY_PATH: str({ devDefault: '../../.secrets/public-key.pem' }),
+  SENTRY_DSN: str({ devDefault: undefined }),
```
If I remember correctly, its by design that you can leave SENTRY_DSN unset. Even in prod environments

naftis committed:
fix: set sentry as undefined in prod env as well

naftis committed:
fix: tests using the wrong env

naftis committed:
fix(tests): explicitly set the required values for prod

this is similar how it works in prod - think it's good to have them set explicit

rikukissa commented on packages/auth/src/features/authenticate/handler.test.ts:
```diff
@@ -71,7 +71,19 @@ describe('authenticate handler receives a request', () => {
     it('generates a mobile verification code and sends it to notification gateway', async () => {
-      server = await createServerWithEnvironment({ NODE_ENV: 'production' })
+      server = await createServerWithEnvironment({
+        NODE_ENV: 'production',
+        AUTH_HOST: '0.0.0.0',
+        AUTH_PORT: '4040',
+        CLIENT_APP_URL: 'http://localhost:3000/',
+        COUNTRY_CONFIG_URL: 'http://localhost:3040/',
+        DOMAIN: '*',
+        LOGIN_URL: 'http://localhost:3020/',
+        METRICS_URL: 'http://localhost:1050',
+        NOTIFICATION_SERVICE_URL: 'http://localhost:2020/',
+        REDIS_HOST: 'localhost',
+        USER_MANAGEMENT_URL: 'http://localhost:3030/'
```
@naftis how about we take the variables from AUTH_HOST to USER_MANAGEMENT_URL to inside `createServerWithEnvironment`?

naftis replied:
I would have wanted to avoid it as we have this:

https://github.com/opencrvs/opencrvs-core/blob/a33c55c407a46bd54e0621c2a980fba966812566/packages/auth/src/features/resend/handler.test.ts#L84

to keep devDefaults only in one place.

I'll create a separate createProductionEnvironment.

naftis committed:
refactor(tests): use a generic handler for production environments

rikukissa committed:
Merge branch 'develop' into ocrvs-5932-auth

rikukissa commented on packages/auth/src/environment.ts:
```diff
@@ -0,0 +1,33 @@
+
+export const env = cleanEnv(process.env, {
+  REDIS_HOST: str({ devDefault: 'localhost' }),
+  AUTH_HOST: str({ devDefault: '0.0.0.0' }),
+  AUTH_PORT: port({ devDefault: 4040 }),
```
```suggestion
  AUTH_HOST: str({ default: '0.0.0.0' }),
  AUTH_PORT: port({ default: 4040 }),
```

naftis committed:
fix: AUTH_HOST and AUTH_PORT can be default

they are rarely being re-configured while this allows for that.

Co-authored-by: Riku Rouvila <riku.rouvila@gmail.com>

rikukissa committed:
Merge branch 'develop' into ocrvs-5932-auth

rikukissa committed:
Merge branch 'develop' into ocrvs-5932-auth
